src/cfnlint/rules/resources/HardCodedArnProperties.py

```diff
@@ -0,0 +1,34 @@
+"""
+Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: MIT-0
+"""
+import re
+from cfnlint.rules import CloudFormationLintRule
+from cfnlint.rules import RuleMatch
+
+
+class HardCodedArnProperties(CloudFormationLintRule):
+    """Checks Resources if ARNs use correctly placed Pseudo Parameters instead of hardcoded Partition, Region, and Account Number"""
+    id = 'W3042'
+    shortdesc = 'ARNs should use correctly placed Pseudo Parameters'
+    description = 'Checks Resources if ARNs use correctly placed Pseudo Parameters instead of hardcoded Partition, Region, and Account Number'
+    source_url = ''
+    tags = ['resources']
+
+    def match(self, cfn):
+        """Check CloudFormation Resources"""
+        matches = []
+        pattern = re.compile('arn:(\$\{[^:]*::[^:]*}|[^:]*):[^:]+:(\$\{[^:]*::[^:]*}|[^:]*):(\$\{[^:]*::[^:]*}|[^:]*)')

```

Comment:
Try using a raw string by adding `r` before the open quotes, i.e. `re.compile(r'arn:(\$\{[^:]*::[^:]*}|[^:]*):[^:]+:(\$\{[^:]*::[^:]*}|[^:]*):(\$\{[^:]*::[^:]*}|[^:]*)')`.